core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -747,11 +747,14 @@ private[spark] object Utils
       hadoopConf: Configuration,
       fileOverwrite: Boolean,
       filename: Option[String] = None): Unit = {
+    if (!fs.exists(path)) {
+      throw new FileNotFoundException(s"$path not found")
+    }
     if (!targetDir.exists() && !targetDir.mkdir()) {
       throw new IOException(s"Failed to create directory ${targetDir.getPath}")
     }
     val dest = new File(targetDir, filename.getOrElse(path.getName))
-    if (SparkHadoopUtil.isFile(fs, path)) {
+    if (fs.getFileStatus(path).isFile) {
```

Comment:
It seems `fs.exists(path) && fs.getFileStatus(path).isFile` have the same semantics as origin `SparkHadoopUtil.isFile(fs, path)`.